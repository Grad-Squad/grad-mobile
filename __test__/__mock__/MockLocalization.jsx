import React from 'react';
import i18n from 'i18n-js';
import { LocalizationContext } from 'localization';
import { childrenPropType } from 'proptypes';

// it loads the localization files when importing the context
const MockLocalization = ({ children }) => (
  <LocalizationContext.Provider
    value={{
      // availableLanguages,
      // setLanguage,
      language: 'en',
      t: i18n.t,
      isRTL: false,
    }}
  >
    {children}
  </LocalizationContext.Provider>
);

MockLocalization.propTypes = {
  children: childrenPropType.isRequired,
};
export default MockLocalization;
