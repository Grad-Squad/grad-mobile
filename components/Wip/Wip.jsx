import React from 'react';
import { Image, StyleSheet, View } from 'react-native';
import EduText from 'common/EduText';
import { Colors } from 'styles';
import { useLocalization } from 'localization';
import { TransparentButton } from 'common/Input/Button';
import { useNavigation } from '@react-navigation/core';

const Wip = () => {
  const { t } = useLocalization();
  const navigation = useNavigation();
  return (
    <View style={styles.background}>
      <Image
        style={styles.wipGif}
        source={require('../../assets/gifs/WIP.gif')}
      />
      <EduText style={styles.header}>{t('WIP/WIP')}</EduText>
      <EduText style={styles.text}>
        {t(
          'WIP/We are currently working on this feature. It will be available as soon as possible.'
        )}
      </EduText>

      <TransparentButton
        text={t('WIP/Go back')}
        onPress={() => navigation.goBack()}
        style={styles.goBackButton}
      />
    </View>
  );
};

Wip.propTypes = {};
Wip.defaultProps = {};

export default Wip;

const styles = StyleSheet.create({
  background: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.background,
  },
  wipGif: {},
  header: {
    fontFamily: 'Lato_700Bold',
    fontSize: 72,
    marginTop: 35,
    marginBottom: 6,
  },
  text: {
    fontSize: 18,
    width: 250,
  },
  goBackButton: {
    position: 'absolute',
    bottom: 10,
    alignSelf: 'center',
  },
});
