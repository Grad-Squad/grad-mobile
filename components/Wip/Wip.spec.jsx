import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import MockLocalization from '__test__/__mock__/MockLocalization';
// eslint-disable-next-line import/no-extraneous-dependencies
import { useNavigation } from '@react-navigation/core';
import { Provider } from 'react-native-paper';
import Wip from './Wip';

jest.mock('@react-navigation/core', () => {
  const goBackMock = jest.fn().mockName('go back');
  return {
    useNavigation: () => ({
      goBack: goBackMock,
    }),
  };
});

let wip;
beforeEach(() => {
  wip = render(
    <Provider>
      <MockLocalization>
        <Wip />
      </MockLocalization>
    </Provider>
  );
  useNavigation().goBack.mockClear();
});

it('should match snapshot', () => {
  expect(wip.toJSON()).toMatchSnapshot();
});

it('should call goBack when Go Back button is pressed', () => {
  fireEvent.press(wip.getByText('Go back'));
  expect(useNavigation().goBack).toHaveBeenCalledTimes(1);
});
