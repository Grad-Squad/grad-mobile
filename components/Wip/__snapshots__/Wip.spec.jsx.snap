// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FBFBFB",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 20,
            "elevation": 3,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 2,
          },
          Object {},
        ]
      }
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Image
            onLoad={[Function]}
            resizeMode="contain"
            source={
              Object {
                "uri": "https://c.tenor.com/DaSh5T93TgUAAAAC/cat-typing.gif",
              }
            }
            style={
              Array [
                Object {
                  "marginBottom": 26.68,
                  "resizeMode": "contain",
                },
                Object {
                  "height": 600,
                  "width": 600,
                },
                Array [
                  Object {
                    "marginBottom": 0,
                  },
                  Object {},
                ],
              ]
            }
          />
        </View>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Lato_400Regular",
          },
          Object {
            "color": "#3C3C3C",
          },
          Object {
            "fontFamily": "Lato_700Bold",
            "fontSize": 72,
            "marginBottom": 6,
            "marginTop": 35,
          },
        ]
      }
    >
      WIP
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Lato_400Regular",
          },
          Object {
            "color": "#3C3C3C",
          },
          Object {
            "fontSize": 18,
            "width": 250,
          },
        ]
      }
    >
      We are currently working on this feature. It will be available as soon as possible.
    </Text>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 9,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "paddingVertical": 7,
          },
          Array [
            Object {},
            false,
            Object {
              "alignSelf": "center",
              "bottom": 10,
              "position": "absolute",
            },
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#3C3C3C",
              "fontFamily": "Lato_400Regular",
              "fontSize": 22,
            },
            Array [
              Object {
                "color": "#0A7DA4",
              },
              false,
              Object {},
            ],
          ]
        }
      >
        Go back
      </Text>
    </View>
  </View>
</View>
`;
